### PR TITLE
UCP/API: improved comments for send/recv nbx API calls

### PR DESCRIPTION
**Pull request by hoopoepg**
- added description for supported flags and values for
  tag_send/recv_nbx calls

### CODE REVIEW

**Comment by hoopoepg** [comment]
@yosefe @brminich are you ok with such doc format?
if yes - I will update comments to add references & actual flags supported

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -2634,6 +2634,15 @@ ucs_status_ptr_t ucp_tag_send_sync_nb(ucp_ep_h ep, const void *buffer, size_t co
  *
  * @note The user should not modify any part of the @a buffer after this
  *       operation is called, until the operation completes.
+ * @note The argument @a param->op_attr_mask should be OR combination of
+ *       flags UCP_OP_ATTR_FIELD_REQUEST, UCP_OP_ATTR_FIELD_CALLBACK,
+ *       UCP_OP_ATTR_FIELD_DATATYPE, UCP_OP_ATTR_FLAG_NO_IMM_CMPL,
+ *       UCP_OP_ATTR_FLAG_NO_ZCOPY and UCP_OP_ATTR_FLAG_FORCE_IMM_CMPL.
+ * @note Combination of flags
+ *       UCP_OP_ATTR_FLAG_NO_IMM_CMPL | UCP_OP_ATTR_FLAG_FORCE_IMM_CMPL is not
```
Comment (brminich) [inline src/ucp/api/ucp.h:2642]: this is a general note for any function. Please describe this once where the flags are defined

File: src/ucp/api/ucp.h [hunk 2]
```diff
@@ -2634,6 +2634,15 @@ ucs_status_ptr_t ucp_tag_send_sync_nb(ucp_ep_h ep, const void *buffer, size_t co
  *
  * @note The user should not modify any part of the @a buffer after this
  *       operation is called, until the operation completes.
+ * @note The argument @a param->op_attr_mask should be OR combination of
```
Comment (brminich) [inline src/ucp/api/ucp.h:2637]: maybe `The argument @a param->op_attr_mask may be combination of any of the following flags: ....`
i'd also add this phrase here: ` For every UCP_OP_ATTR_FIELD.. flag the corresponding @param structure field must be defined.`
The you will not need to explain param fields separately for every operation

**Comment by yosefe** [comment]
@hoopoepg is this PR still relevant?

**Pull request closed without merging**